bukkit/src/main/java/me/lucko/luckperms/bukkit/migration/MigrationPermissionsEx.java

```diff
@@ -99,11 +106,34 @@ public CommandResult execute(LuckPermsPlugin plugin, Sender sender, Object o, Li
             // migrate data
             migrateEntity(group, lpGroup, groupWeight);
 
-            plugin.getStorage().saveGroup(lpGroup);
+            // remember known ladders
+            if (group.isRanked() && !ladders.contains(group.getRankLadder().toLowerCase())) {
+                ladders.add(group.getRankLadder().toLowerCase());
+            }
+
+            plugin.getStorage().saveGroup(lpGroup).join();
             log.logAllProgress("Migrated {} groups so far.", groupCount.incrementAndGet());
         });
         log.log("Migrated " + groupCount.get() + " groups");
 
+        // Migrate all ladders/tracks.
+        log.log("Starting tracks migration.");
+        for (String rankLadder : ladders) {
+            plugin.getStorage().createAndLoadTrack(rankLadder, CreationCause.INTERNAL).join();
+            Track track = plugin.getTrackManager().getIfLoaded(rankLadder);
+
+            // Get a list of all groups in a ladder
+            List<String> ladder = new TreeMap<Integer, PermissionGroup>(manager.getRankLadder(rankLadder)).values().stream().map(ladderGroup -> MigrationUtils.standardizeName(ladderGroup.getName())).collect(Collectors.toList());
```

Comment:
This is still quite a convoluted implementation.

I'd suggest something more like:

```java
List<String> ladder = manager.getRankLadder("test").entrySet().stream()
        .sorted(Comparator.<Map.Entry<Integer, PermissionGroup>>comparingInt(e -> e.getKey()).reversed())
        .map(e -> MigrationUtils.standardizeName(e.getValue().getName()))
        .collect(Collectors.toList());

track.setGroups(ladder);
plugin.getStorage().saveTrack(track);
```